src/python/WMCore/ReqMgr/DataStructs/Request.py

```diff
@@ -188,8 +188,12 @@ def andFilterCheck(self, filterDict):
             if not isinstance(value, list):
                 value = [value]
 
-            if key in self.data:
-                if self.data[key] not in value:
+            reqValue = self.get(key)
```

Comment:
actually self.get(key) is correct. so it can get the Task/StepChain param correctly.